src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMapping.java

```diff
@@ -140,6 +160,22 @@ public static List<HearingLocation> getHearingLocations(SscsCaseData caseData,
             .collect(Collectors.toCollection(ArrayList::new));
     }
 
+    private static String getVenueID(SscsCaseData caseData, String nextHearingVenue) {
+        if (SOMEWHERE_ELSE.equals(nextHearingVenue)) {
+            return caseData.getAdjournCaseNextHearingVenueSelected().getValue().getCode();
+        } else if (SAME_VENUE.equals(nextHearingVenue)) {
+            Hearing latestHearing = caseData.getLatestHearing();
+
+            if (nonNull(latestHearing)) {
+                return latestHearing.getValue().getVenueId();
+            } else {
+                throw new IllegalStateException("Failed to determine next hearing venue due to no latest hearing");
+            }
+        } else {
+            throw new IllegalStateException("Failed to determine next hearing venue: " + nextHearingVenue);
+        }
```

Comment:
As above I would suggest not having an else if as you are returning.
And I would also suggest not worring about checking for SAME_VENUE, you already blank checking on the parent method and there are only two options in the fixed listed.
```suggestion
        } 
        
        Hearing latestHearing = caseData.getLatestHearing();

        if (nonNull(latestHearing)) {
            return latestHearing.getValue().getVenueId();
        } else {
            throw new InvalidMappingException("Failed to determine next hearing venue due to no latest hearing");
       }
```

Reply:
Done